sos/plugins/sapnw.py

```diff
@@ -32,13 +32,13 @@ def collect_list_instances(self):
             "/usr/sap/hostctrl/exe/saphostctrl -function ListInstances",
             suggest_filename="SAPInstances"
         )
-        if not inst_out:
+        if inst_out['status'] != 0:
             return
 
         sidsunique = set()
         # Cycle through all the instances, get 'sid', 'instance_number'
         # and 'vhost' to determine the proper profile
-        p = open(inst_out, "r").read().splitlines()
+        p = open(inst_out['filename'], "r").read().splitlines()
```

Comment:
But `collect_cmd_output()` also returns an `'output'` entry in the dictionary - we should just `splitlines()` on that and use it, rather than re-read the content into memory a second time.